src/nunit.analyzers/UseAssertThatAsync/UseAssertThatAsyncAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using NUnit.Analyzers.Constants;
+
+namespace NUnit.Analyzers.UseAssertThatAsync;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class UseAssertThatAsyncAnalyzer : BaseAssertionAnalyzer
+{
+    private static readonly string[] firstParameterCandidates =
+    {
+        NUnitFrameworkConstants.NameOfActualParameter,
+        NUnitFrameworkConstants.NameOfConditionParameter,
+    };
+
+    private static readonly DiagnosticDescriptor descriptor = DiagnosticDescriptorCreator.Create(
+        id: AnalyzerIdentifiers.UseAssertThatAsync,
+        title: UseAssertThatAsyncConstants.Title,
+        messageFormat: UseAssertThatAsyncConstants.Message,
+        category: Categories.Assertion,
+        defaultSeverity: DiagnosticSeverity.Info,
+        description: UseAssertThatAsyncConstants.Description);
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(descriptor);
+
+    protected override void AnalyzeAssertInvocation(Version nunitVersion, OperationAnalysisContext context, IInvocationOperation assertOperation)
+    {
+        // Assert.ThatAsync was introduced in NUnit 4
+        if (nunitVersion.Major < 4)
+            return;
+
+        if (assertOperation.TargetMethod.Name != NUnitFrameworkConstants.NameOfAssertThat)
+            return;
+
+        var arguments = assertOperation.Arguments
+            .Where(a => a.ArgumentKind == ArgumentKind.Explicit) // filter out arguments that were not explicitly passed in
+            .ToArray();
+
+        // The first parameter is usually the "actual" parameter, but sometimes it's the "condition" parameter.
```

Comment:
The only argument name we want to catch is actually called `del` as used in:
` public static void That<TActual>(ActualValueDelegate<TActual> del, IResolveConstraint expr`